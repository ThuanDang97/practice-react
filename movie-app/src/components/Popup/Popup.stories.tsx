import { Story } from '@storybook/react';
import Popup from './Popup';
import { TProps } from './Popup';

export default {
  title: 'Components/Popup',
  component: Popup,
};

const Template: Story<TProps> = (args): JSX.Element => (
  <div
    style={{
      background: 'gray',
      width: '15%',
    }}
  >
    <Popup {...args} />
  </div>
);

export const Basic: Story<TProps> = Template.bind({});
Basic.args = {
  title: 'Element',
  src: 'http://localhost:3000/src/assets/heart-black.svg',
  alt: 'Element',
  color: 'black',
};

export const Active: Story<TProps> = Template.bind({});
Active.args = {
  title: 'Element',
  src: 'http://localhost:3000/src/assets/heart-black.svg',
  alt: 'Element',
  color: 'pink',
};
