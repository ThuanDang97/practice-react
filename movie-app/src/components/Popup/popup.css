.group {
  padding: 10px 20px;
  display: flex;
  cursor: pointer;
}

.group i {
  width: 15px;
  height: 15px;
  margin-right: 5px;
  padding-top: 2px;
}

.group:hover {
  color: white;
}

.group:hover .black {
  filter: brightness(0) invert(1);
}

.pink {
  filter: invert(51%) sepia(41%) saturate(2511%) hue-rotate(289deg) brightness(92%)
    contrast(87%);
}
