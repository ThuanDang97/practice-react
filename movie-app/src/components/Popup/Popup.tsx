import { memo, MouseEvent } from 'react';
import './popup.css';

export type TProps = {
  src: string;
  alt: string;
  title: string;
  color: 'black' | 'pink';
  onHandleClick: (e: MouseEvent<HTMLDivElement>) => void;
};

const Popup = (props: TProps): JSX.Element => {
  const { src, alt, title, color, onHandleClick } = props;

  return (
    <div className="group" onClick={onHandleClick} data-testid="popup">
      <i className={color}>
        <img src={src} alt={alt} className="black" />
      </i>
      {title}
    </div>
  );
};

export default memo(Popup);
