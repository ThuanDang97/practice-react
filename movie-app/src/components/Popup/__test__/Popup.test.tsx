import { screen, render } from '@testing-library/react';
import { setup } from '../../../support/Setup';
import Popup, { TProps } from '../Popup';

describe('Test Popup component', () => {
  const defaultProps: TProps = {
    src: '',
    alt: '',
    title: 'Element',
    color: 'black',
    onHandleClick: jest.fn(),
  };

  test('Component Popup matches DOM Snapshot', () => {
    const component = <Popup {...defaultProps} />;

    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Popup should render correctly', () => {
    setup(<Popup {...defaultProps} />);

    expect(screen.getByTestId('popup')).toHaveTextContent('Element');
  });

  test('Component Popup should called handleClick', async () => {
    const { user } = setup(<Popup {...defaultProps} />);

    await user.click(screen.getByTestId('popup'));
    expect(defaultProps.onHandleClick).toHaveBeenCalled();
  });
});
