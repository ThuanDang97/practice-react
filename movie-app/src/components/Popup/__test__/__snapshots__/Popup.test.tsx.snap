// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Popup component Component Popup matches DOM Snapshot 1`] = `
<div>
  <div
    class="group"
    data-testid="popup"
  >
    <i
      class="black"
    >
      <img
        alt=""
        class="black"
        src=""
      />
    </i>
    Element
  </div>
</div>
`;
